src/hooks/useResizeObserver.ts

```diff
@@ -9,26 +9,29 @@ import { useCallback, useEffect, useRef, useState } from 'react';
  * - height - Height of the container
  */
 export const useResizeObserver = () => {
-  const resizeObserver = useRef<ResizeObserver>();
+  const [node, setNode] = useState<HTMLElement | null>(null);
   const [width, setWidth] = useState<number>();
   const [height, setHeight] = useState<number>();
 
   const ref = useCallback((node: HTMLElement | null) => {
+    setNode(node);
+  }, []);
+
+  useLayoutEffect(() => {
     if (!node) return;
-    resizeObserver.current = new ResizeObserver(() => {
+    const observer = new ResizeObserver((entries) => {
       window.requestAnimationFrame(() => {
-        setWidth(node.offsetWidth);
-        setHeight(node.offsetHeight);
+        entries.forEach((entry) => {
+          setWidth(entry.target.clientWidth);
+          setHeight(entry.target.clientHeight);
+        });
```

Comment:
for multiple entries, single width and height state is updated (there should be probably only one entry, but it looks weird)